doc-generator/src/main/scala/DocGenerator.scala

```diff
@@ -0,0 +1,95 @@
+import com.codacy.plugins.api._
+import com.codacy.plugins.api.results._
+import play.api.libs.json.Json
+import scala.xml._
+
+object DocGenerator extends App {
+  val scalastyleDefinition = XML.loadString(os.read(os.resource / "scalastyle_definition.xml"))
+  val scalastyleDocumentation = XML.loadString(os.read(os.resource / "scalastyle_documentation.xml"))
+
+  case class Checker(id: String, defaultLevel: String, cls: String, parameters: Seq[Checker.Parameter]) {
+    def patternId: String = cls.split('.').last
+
+    // Split camel case patternId into separate words
+    def title: String = patternId.stripSuffix("Checker").replaceAll("([a-z](?=[A-Z]))", "$1 ")
```

Comment:
Does this write "$1 "? Is it what we want? or the replace all interprets that string to be the first capture group?

Reply:
Actually I took this code from [here](https://stackoverflow.com/a/18379502/5910054) 
I didn't study how it does work 🙈

Reply:
You copied code from a JS link to Scala and don't know how to justify it? :O please don't xD
Will wait for an explanation or a solution that you understand :D 

Also, there is not a lihaoyi.string.splitCamelCase? 😂

Reply:
> You copied code from a JS link to Scala and don't know how to justify it? :O please don't xD
Will wait for an explanation or a solution that you understand :D

Well, I converted it to Scala 😎 🤣

> Also, there is not a lihaoyi.string.splitCamelCase? 😂

Maybe there is 😄

Reply:
> Does this write "$1 "? Is it what we want? or the replace all interprets that string to be the first capture group?

Studied it. What it does is to replace all the lower case characters followed by an upper case char \(checked with a [positive lookahead](https://stackoverflow.com/questions/1570896/what-does-mean-in-a-regular-expression)\) with the char followed by a space.

For example: in `CamelCaseString` it replaces the `l` and second `e` ( the only lowercase chars followed by an upper case one with `l ` and `e `, returning `Camel Case String`

Reply:
NICE! thx! :D